src/utils/quote_report_utils.ts

```diff
@@ -123,7 +132,8 @@ export const quoteReportUtils = {
             const extendedQuoteReport: ExtendedQuoteReport = {
                 quoteId: logOpts.quoteId,
                 taker: logOpts.taker,
-                timestamp: Date.now(),
+                timestamp:
+                    logOpts.submissionBy === 'taker' ? getTimestampFromUniqueId(logOpts.decodedUniqueId) : Date.now(),
```

Comment:
If `submissionBy` is not `taker`, a single `decodedUniqueId` will not be enough to refer to a Swap quote report, right?

Reply:
Seems `submissionBy` is always `taker` when this function is called. However in type definition it could be `metaTxn`. @Ktl-XV If interface `ExtendedQuoteReportForMetaTxn` is never used, could we delete it?

Reply:
Yeah, we can remove it

Reply:
Oh only call to `/meta_transaction/v1` will result in MetaTxn type of quote report. As long as we are calling `/swap/v1` we should always have a decodedUniqueId.